### PR TITLE
chore(codepipeline): add missing action for EcsDeployAction

### PR DESCRIPTION
**Pull request by pahud**
### Issue # (if applicable)

Closes https://github.com/aws/aws-cdk/issues/29400

### Reason for this change

Missing required action as described from the [doc](https://docs.aws.amazon.com/codepipeline/latest/userguide/security-iam.html#how-to-custom-role).

### Checklist
- [x] My code adheres to the [CONTRIBUTING GUIDE](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md) and [DESIGN GUIDELINES](https://github.com/aws/aws-cdk/blob/main/docs/DESIGN_GUIDELINES.md)

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/aws-cdk-lib/aws-codepipeline-actions/lib/ecs/deploy-action.ts [hunk 1]
```diff
@@ -80,14 +80,15 @@ export class EcsDeployAction extends Action {
   protected bound(_scope: Construct, _stage: codepipeline.IStage, options: codepipeline.ActionBindOptions):
   codepipeline.ActionConfig {
     // permissions based on CodePipeline documentation:
-    // https://docs.aws.amazon.com/codepipeline/latest/userguide/how-to-custom-role.html#how-to-update-role-new-services
-    options.role.addToPolicy(new iam.PolicyStatement({
+    // https://docs.aws.amazon.com/codepipeline/latest/userguide/security-iam.html#how-to-custom-role
+    options.role.addToPrincipalPolicy(new iam.PolicyStatement({
```
Comment (paulhcsun) [inline packages/aws-cdk-lib/aws-codepipeline-actions/lib/ecs/deploy-action.ts:84]: Question: what's the difference between `addToPolicy` and `addToPrincipalPolicy` and why is it needed if we're just filling in the missing action?

File: packages/aws-cdk-lib/aws-codepipeline-actions/lib/ecs/deploy-action.ts [hunk 2]
```diff
@@ -80,14 +80,15 @@ export class EcsDeployAction extends Action {
   protected bound(_scope: Construct, _stage: codepipeline.IStage, options: codepipeline.ActionBindOptions):
   codepipeline.ActionConfig {
     // permissions based on CodePipeline documentation:
-    // https://docs.aws.amazon.com/codepipeline/latest/userguide/how-to-custom-role.html#how-to-update-role-new-services
```
Comment (paulhcsun) [inline packages/aws-cdk-lib/aws-codepipeline-actions/lib/ecs/deploy-action.ts:83]: Is the link change just to point to the more specific section?

File: packages/aws-cdk-lib/aws-codepipeline-actions/lib/ecs/deploy-action.ts (hunk 1 above)
Comment (pahud) [reply packages/aws-cdk-lib/aws-codepipeline-actions/lib/ecs/deploy-action.ts:84]: https://github.com/aws/aws-cdk/blob/ba41996939a4c0b06e954c7de7232adef4ce78f5/packages/aws-cdk-lib/aws-iam/lib/principals.ts#L59-L73

`addToPolicy` is deprecated and should use `addToPrincipalPolicy` instead. But I agree we should leave it as is for consistency. I will change it back to `addToPolicy`.

File: packages/aws-cdk-lib/aws-codepipeline-actions/lib/ecs/deploy-action.ts (hunk 2 above)
Comment (pahud) [reply packages/aws-cdk-lib/aws-codepipeline-actions/lib/ecs/deploy-action.ts:83]: The previous link is invalid and redirect to the welcome page.

**Comment by pahud** [comment]
@paulhcsun  All addressed please take a look again.

**Review by paulhcsun — APPROVED** [review]
Looks good, thanks @pahud!

**Pull request merged**